### PR TITLE
remove /reference pages + reducers on the frontend

### PR DESCRIPTION
**Pull request by kdoh**
💬  [Slack Discussion](https://metaboat.slack.com/archives/C01NT034X2B/p1683826153102389)

EOL the reference pages at /reference that you probably didn't even know we had.

The data reference was an experiment in trying to add context and help with data exploration. I think it's safe to say we've since updated our model of how we think this kind of information can be leveraged to help you understand your data and at this point there's no need to keep shipping this product and code to our users.

Todos
- [x] Update / adjust e2e tests based on conversation with @nemanjaglumac

### CODE REVIEW

**Comment by kulyk** [comment]
Nice work! Can't wait to see our coverage increase with -7K LoC :D

**Comment by iethree** [comment]
Will this involve dropping tables from the App DB? could we get a backend engineer to evaluate the backend blast radius for this removal? Maybe @dpsutton could take a quick look, or delegate to someone else to do so?

**Comment by kdoh** [comment]
@nemanjaglumac Commit https://github.com/metabase/metabase/pull/30709/commits/264bac1dcef614d1899c20af5242dd468c280bd6 has a bunch of questions I'd like to talk through via PR comments about ways to make sure we still have coverage for the way the reference was used to verify behavior. Would appreciate your thoughts if you're able to suggest some basic patterns or approaches. I removed any obvious cases where there was just a navigation to the reference and a check for text that seemed independent of a larger workflow.

There's specifically a pattern for the `admin/datamodel/*` tests that seemed to use the admin panel to make a change, and then go verify things like still being able to edit via the reference that I'm curious about updating.  In a case like [this tet](https://github.com/metabase/metabase/commit/264bac1dcef614d1899c20af5242dd468c280bd6#diff-a4ac9e079a2c22f25415972b3a40be0c56bc84023c7e570470801c6e7270bfa1L69) wasn't the behavior verified before even visiting /reference? Is there a need to check twice or am I missing something?

**Review by ranquild — APPROVED** [review]

File: e2e/test/scenarios/admin/datamodel/editor.cy.spec.js [hunk 1]
```diff
@@ -58,19 +58,14 @@ describe("scenarios > admin > datamodel > editor", () => {
       });
     });
 
+    // QUESTION - can we check update in the admin instead?
```
Comment (kulyk) [inline e2e/test/scenarios/admin/datamodel/editor.cy.spec.js:61]: Think this comment is no longer relevant?

**Review by kulyk — APPROVED** [review]
Fantastic

**Comment by lvgiacomin** [comment]
Hey guys, just to give my opinion. The reference page is kind of hidden on metabase, so it is not easy to find. But without a Metrics reference (or a Metrics Catalog) it is considerably harder for non-technical users to self-serve, and even for more technical users. They will need to know in which table the metric is created, and then make a question. As a reference, [Transform](https://transform.co/) has a Metric painel, and then users can start to explore data from that.

**Pull request merged**